packages/text-field/index.js

```diff
@@ -0,0 +1,367 @@
+        {trailingIcon ? this.renderIcon(trailingIcon) : null}
+      </div>
+    );
+
+    if (helperText) {
+      return ([
+        textField, this.renderHelperText(),
+      ]);
+    }
+    return textField;
+  }
+
+  renderInput() {
+    const child = React.Children.only(this.props.children);
+    const props = this.inputProps(child.props);
+    return React.cloneElement(child, props);
+  }
+
+  renderLabel() {
+    const {label, floatingLabelClassName} = this.props;
+    const {inputId} = this.state;
+    return (
+      <FloatingLabel
+        className={floatingLabelClassName}
+        float={this.state.shouldFloatLabel}
+        handleWidthChange={
+          (labelWidth) => this.setState({labelWidth})}
+        ref={this.floatingLabelElement}
+        htmlFor={inputId}
+      >
+        {label}
+      </FloatingLabel>
+    );
+  }
+
+  renderLineRipple() {
+    const {lineRippleClassName} = this.props;
+    const {activeLineRipple, lineRippleCenter} = this.state;
+    return (
+      <LineRipple
+        rippleCenter={lineRippleCenter}
+        className={lineRippleClassName}
+        active={activeLineRipple}
+      />
+    );
+  }
+
+  renderNotchedOutline() {
+    const {notchedOutlineClassName} = this.props;
+    const {shouldNotchOutline, labelWidth} = this.state;
+    return (
+      <NotchedOutline
+        className={notchedOutlineClassName}
+        isRtl={this.getIsRtl()}
+        notch={shouldNotchOutline}
+        notchWidth={labelWidth}
+      />
+    );
+  }
+
+  renderHelperText() {
+    const {helperTextAriaHidden, helperTextIsValidationMessage, helperText,
+      helperTextClassName, helperTextPersistent, helperTextRole} = this.props;
+    const {isValid, helperTextShowToScreenReader} = this.state;
+    return (
+      <HelperText
+        className={helperTextClassName}
+        showToScreenReader={helperTextShowToScreenReader}
+        isValid={isValid}
+        role={helperTextRole}
+        key='text-field-helper-text'
+        aria-hidden={helperTextAriaHidden}
+        persistent={helperTextPersistent}
+        isValidationMessage={helperTextIsValidationMessage}
+      >
+        {helperText}
+      </HelperText>
+    );
+  }
+
+  renderIcon(icon) {
+    const {disabled} = this.state;
+    // Toggling disabled will trigger icon.foundation.setDisabled()
+    return (
+      <Icon disabled={disabled}>
+        {icon}
+      </Icon>
+    );
+  }
+}
+
+TextField.propTypes = {
+  'box': PropTypes.bool,
+  'children.props': PropTypes.shape(Input.propTypes),
+  'children': PropTypes.element,
+  'className': PropTypes.string,
+  'dense': PropTypes.bool,
+  'floatingLabelClassName': PropTypes.string,
+  'fullWidth': PropTypes.bool,
+  'helperText': PropTypes.string,
+  'helperTextAriaHidden': PropTypes.bool,
+  'helperTextClassName': PropTypes.string,
+  'helperTextIsValidationMessage': PropTypes.bool,
+  'helperTextPersistent': PropTypes.bool,
+  'helperTextRole': PropTypes.string,
```

Comment:
Should we consider having the end developer pass us a full Helper text component? Instead of these 6 props?

Reply:
I think so...good idea!